x-pack/plugins/apm/server/routes/agent_explorer/fetch_agents_latest_version.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Logger } from '@kbn/core/server';
+import { isEmpty } from 'lodash';
+import fetch from 'node-fetch';
+import {
+  ElasticApmAgentLatestVersion,
+  OtelAgentLatestVersion,
+} from '../../../common/agent_explorer';
+import { AgentName } from '../../../typings/es_schemas/ui/fields/agent';
+import { ErrorWithStatusCode } from './error_with_status_code';
+
+export interface AgentLatestVersionsResponse {
+  data: AgentLatestVersions;
+  error?: { message: string; type?: string; statusCode?: string };
+}
+
+type AgentLatestVersions = Record<
+  AgentName,
+  ElasticApmAgentLatestVersion | OtelAgentLatestVersion
+>;
+
+export const fetchAgentsLatestVersion = async (
+  logger: Logger,
+  latestAgentVersionsUrl: string
+): Promise<AgentLatestVersionsResponse> => {
+  if (isEmpty(latestAgentVersionsUrl)) {
```

Comment:
What about a `Boom.badRequest` here? I don't think this API should be called if `latestAgentVersionsUrl` is not defined. Or maybe `Boom.notImplemented`.

Reply:
It's not being called form the UI, but I put this check just in case. I will check what you are suggesting, I think it's good idea